vendor/bat-native-ledger/src/bat/ledger/internal/database/database_unblinded_token.cc

```diff
@@ -521,6 +539,155 @@ void DatabaseUnblindedToken::MarkRecordListAsSpent(
   ledger_->RunDBTransaction(std::move(transaction), transaction_callback);
 }
 
+void DatabaseUnblindedToken::MarkRecordListAsReserved(
+    const std::vector<std::string>& ids,
+    const std::string& redeem_id,
+    ledger::ResultCallback callback) {
+  if (ids.empty()) {
+    BLOG(1, "List of ids is empty");
+    callback(ledger::Result::LEDGER_ERROR);
+    return;
+  }
+
+  auto transaction = ledger::DBTransaction::New();
+
+  const std::string id_values = GenerateStringInCase(ids);
+
+  std::string query = base::StringPrintf(
+      "UPDATE %s SET redeem_id = ?, reserved_at = ? "
+      "WHERE ( "
+        "SELECT COUNT(*) FROM %s "
```

Comment:
What is the purpose of this subquery in the filter?

Reply:
I wanted to ensure that if (say) 3 records need to be updated then exactly that number were updated.

Reply:
Ah, kind of like an assertion check. Seems fine.